snapm/manager/_boot.py

```diff
@@ -53,6 +58,123 @@
 _DEV_PREFIX = "/dev/"
 
 
+class Fstab:
+    """
+    A class to read and query data from an fstab file.
+
+    This class reads an fstab-like file and provides methods to iterate
+    over its entries and look up specific entries based on their properties.
+
+    Attributes:
+        path (str): The path to the fstab file being read.
+        entries (list): A list of named tuples, where each tuple represents an entry.
+    """
+
+    # Define a named tuple to give structure to each fstab entry.
+    # This makes the code more readable than using a regular tuple.
+    FstabEntry = collections.namedtuple(
+        'FstabEntry',
+        ['what', 'where', 'fstype', 'options', 'freq', 'passno']
+    )
+
+    def __init__(self, path='/etc/fstab'):
+        """
+        Initializes the Fstab object by reading and parsing the file.
+
+        Args:
+            path (str, optional): The path to the fstab file.
+                                  Defaults to '/etc/fstab'.
+
+        Raises:
+            FileNotFoundError: If the specified path does not exist.
+            IOError: If there is an error reading the file.
+        """
+        self.path = path
+        self.entries = []
+        self._read_fstab()
+
+    def _read_fstab(self):
+        """
+        Private method to read and parse the fstab file.
+        It populates the self.entries list.
+        """
+        try:
+            with open(self.path, 'r', encoding="utf8") as f:
+                for line in f:
+                    # 1. Strip leading/trailing whitespace.
+                    line = line.strip()
+
+                    # 2. Ignore empty lines and comments.
+                    if not line or line.startswith('#'):
+                        continue
+
+                    # 3. Split the line into parts.
+                    parts = line.split()
+
+                    # 4. An fstab entry must have 6 fields.
+                    if len(parts) != 6:
+                        # You might want to log a warning here in a real application.
+                        continue
+
+                    # 5. Convert freq and passno to integers for proper typing.
+                    try:
+                        parts[4] = int(parts[4]) # freq
+                        parts[5] = int(parts[5]) # passno
+                    except KeyError:
+                        continue
+
+                    # 6. Create a named tuple and append it to our list.
+                    entry = self.FstabEntry(*parts)
+                    self.entries.append(entry)
+
+        except FileNotFoundError:
+            _log_error(f"Error: The file '{self.path}' was not found.")
```

Comment:
We use lazy formatting for logging (`pylint` complains about it otherwise):

```
W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
```
```
_log_error("Error: the file %s was not found", self.path)
```

That way, the log subsystem can elide the formatting calls if the log message is not selected for output.